lib/arel-helpers/join_association.rb

```diff
@@ -248,6 +250,50 @@ def join_association_6_0_0(table, association, join_type, options = {})
         end
       end
 
+      def join_association_6_1_0(table, association, join_type, options = {})
+        aliases = options.fetch(:aliases, []).index_by(&:table_name)
```

Comment:
Can we convert `aliases` arrays into hashes in the methods above for earlier Rails versions?

Reply:
Will do.